pkg/cmd/roachtest/tests/mixed_version_job_compatibility_in_declarative_schema_changer.go

```diff
@@ -152,7 +165,7 @@ func runDeclarativeSchemaChangerJobCompatibilityInMixedVersion(
 	ctx context.Context, t test.Test, c cluster.Cluster,
 ) {
 	mvt := mixedversion.NewTest(
-		ctx, t, t.L(), c, c.All(), mixedversion.NumUpgrades(1),
+		ctx, t, t.L(), c, c.All(), mixedversion.NumUpgrades(1), mixedversion.MinimumSupportedVersion("v24.1.0"),
```

Comment:
create database, sequences are both guarded by `isV241Active`. we set `use_declarative_schema_changer = unsafe_always` in this test[^1]; so, i figured this was a good use case for `mixedversion.MinimumSupportedVersion` wdyt @rafiss 

[^1]: https://github.com/cockroachdb/cockroach/blob/b74d6dbacaeac1b6f274cdd54513d1ebc754ac5b/pkg/cmd/roachtest/tests/mixed_version_job_compatibility_in_declarative_schema_changer.go#L187-L190

Reply:
I'm confused by this one -- since master is already 24.2, the existing `NumUpgrades(1)` option makes `MinimumSupportedVersion("v24.1.0")` a bit of a no-op, no?

Another thing to keep in mind is that in the not-so-distant-future, mixedversion tests will start performing "skip-version" upgrades. Forcing the min-version to 24.1 would opt this test out of skip-version upgrades (23.2 -> 24.2), which I don't think any test should do.

Reply:
> which I don't think any test should do

Although, if I remember correctly, this test could be doing things very specific to the exact previous release, so maybe not doing skip-version upgrades in this test might be the only option.

Reply:
> since master is already 24.2

at the time i created this PR, master didn't have 24.2 changes & i incorrectly assumed that this `MinimumSupportedVersion` was the part of the puzzle that was missing (iirc, tests were going from 23.2-24.1)

i marked this ready for review bc 24.2 is now in, but yes i should definetely get rid of `MinimumSupportedVersion` -- thanks for the catch and the insight!

> this test could be doing things very specific to the exact previous release, so maybe not doing skip-version upgrades in this test might be the only option
yes, this test checks job compatability between the currecnt release and the exact previous release